src/events.rs

```diff
@@ -65,6 +77,10 @@ impl EventQueue {
 		drop(queue);
 
 		if should_notify {
+			if let Some(waker) = self.waker.lock().unwrap().take() {
+				waker.wake();
```

Comment:
this feels a little weird, can you help me understand? is this some weird edge-case where the user is using _both_ the sync `wait_next_event` and async in tandem somehow / for some reason?